server/src/internalClusterTest/java/org/elasticsearch/search/aggregations/bucket/terms/StringTermsIT.java

```diff
@@ -1389,16 +1388,20 @@ public void testScriptWithValueType() throws Exception {
             assertThat(terms.getBuckets().size(), equalTo(1));
         }
 
-        String invalidValueType = source.replaceAll("\"value_type\":\"n.*\"", "\"value_type\":\"foobar\"");
+        String invalidValueType = source.replaceAll("\"value_type\":\"n.*?\"", "\"value_type\":\"foobar\"");
 
         try (XContentParser parser = createParser(JsonXContent.jsonXContent, invalidValueType)) {
-            XContentParseException ex = expectThrows(
-                XContentParseException.class,
+            SearchPhaseExecutionException ex = expectThrows(
+                SearchPhaseExecutionException.class,
```

Comment:
We no longer catch invalid value types at parse time.  We do still catch them before aggregation execution though (at ValuesSourceConfig#Resolve time).